src/Mono.Android/java/mono/android/Runtime.java

```diff
@@ -1,12 +1,25 @@
 package mono.android;
 
 public class Runtime {
+	static java.lang.Class java_lang_Class;
+	static java.lang.Class java_lang_System;
+	static java.lang.Class java_util_TimeZone;
+	static java.lang.Class mono_android_IGCUserPeer;
+	static java.lang.Class mono_android_GCUserPeer;
+
+	static {
+		java_lang_Class = java.lang.Class.class;
+		java_lang_System = java.lang.System.class;
+		java_util_TimeZone = java.util.TimeZone.class;
+		mono_android_IGCUserPeer = mono.android.IGCUserPeer.class;
+		mono_android_GCUserPeer = mono.android.GCUserPeer.class;
+	}
 
 	private Runtime ()
 	{
 	}
 
-	public static native void init (String lang, String[] runtimeApks, String runtimeDataDir, String[] appDirs, ClassLoader loader, String[] externalStorageDirs, String[] assemblies, String packageName);
+	public static native void init (String lang, String[] runtimeApks, String runtimeDataDir, String[] appDirs, ClassLoader loader, String[] externalStorageDirs, String[] assemblies, String packageName, int apiLevel, String[] environmentVariables);
```

Comment:
We are seriously hitting the point of saying, "wow, that's a lot of parameters!"

I assume that parameter passing is faster than reading a static field from a class, but...*is it*?

Is there some point where we should reconsider this signature?

Reply:
Oh, I considered making this signature less horrible, but the register argument won. On ARM we still fit the arguments in registers and that's fast. We should definitely think of some better way of passing those parameters without making the code much slower. For now, I think, it works.

Reply:
We're at 12 parameters (including the `JNIEnv* env, jobject klass` parameters), and ARM has 16 registers, so we have room for 4 more parameters?  Guess that buys us *some* time...